install.sh

```diff
@@ -0,0 +1,182 @@
+CONSUL_HCL_TEMPLATE='data_dir = "/srv/consul/data/"
+log_level = "DEBUG"
+datacenter = "dc1"
+ui = true
+bind_addr = "@BIND_ADDR@"
+client_addr = "0.0.0.0"
+retry_join = ["@JOIN_ADDR@"]'
+
+CONSUL_SERVICE_NAME="consul.service"
+CONSUL_SERVICE_TEMPLATE='[Unit]
+Description="HashiCorp Consul - A service mesh solution"
+Documentation=https://www.consul.io/
+Requires=network-online.target
+After=network-online.target
+ConditionFileNotEmpty=/srv/consul/consul.d/consul.hcl
+PartOf=trento-agent.service
+
+[Service]
+ExecStart=/srv/consul/consul agent -config-dir=/srv/consul/consul.d
+ExecReload=/bin/kill --signal HUP $MAINPID
+KillMode=process
+Restart=on-failure
+RestartSec=5
+Type=notify
+
+
+[Install]
+WantedBy=multi-user.target'
+
+. /etc/os-release
+if [[ ! $PRETTY_NAME =~ "SUSE" ]]; then
+    echo "Operating system is not supported. Exiting."
+    exit 1
+fi
+
+echo "Installing trento-agent..."
+
+function install_deps() {
+    echo "* Installing dependencies... "
+    if ! which unzip > /dev/null 2>&1; then
+        echo "* Installing unzip"
+        zypper in -y unzip > /dev/null
+    fi
+    if ! which curl > /dev/null 2>&1; then
+        echo "* Installing curl"
+        zypper in -y curl > /dev/null
+    fi
+}
+
+function install_consul() {
+    mkdir -p $CONFIG_PATH
+    pushd -- "$CONSUL_PATH" > /dev/null
+    curl -f -sS -O -L "https://releases.hashicorp.com/consul/$CONSUL_VERSION/consul_${CONSUL_VERSION}_linux_amd64.zip" > /dev/null
+    unzip -o "consul_${CONSUL_VERSION}_linux_amd64".zip > /dev/null
+    rm "consul_${CONSUL_VERSION}_linux_amd64".zip
+    popd > /dev/null
+}
+
+function setup_consul() {
+    echo "$CONSUL_HCL_TEMPLATE" |
+        sed "s|@JOIN_ADDR@|${SERVER_IP}|g" |
+        sed "s|@BIND_ADDR@|${AGENT_BIND_IP}|g" |
+        sed "s|@NODE_NAME@|${NODE_NAME}|g" \
+            >${CONFIG_PATH}/consul.hcl
+
+    if [ -f "/usr/lib/systemd/system/$CONSUL_SERVICE_NAME" ]; then
+        echo "  Warning: Consul systemd unit already installed. Removing..."
+        systemctl stop "$CONSUL_SERVICE_NAME"
+        rm "/usr/lib/systemd/system/$CONSUL_SERVICE_NAME"
+    fi
+
+    echo "$CONSUL_SERVICE_TEMPLATE" >/usr/lib/systemd/system/$CONSUL_SERVICE_NAME
+    systemctl daemon-reload
+}
+
+function install_trento() {
+    rpm --import "${TRENTO_REPO_KEY}" > /dev/null
+    path=${TRENTO_REPO%/*}/
+    if zypper lr --details | cut -d'|' -f9 | grep "$path" > /dev/null 2>&1; then
+        echo "* $path repository already exists. Skipping."
+    else
+        zypper ar "$TRENTO_REPO" > /dev/null
+    fi
+    zypper ref > /dev/null
+    if which trento > /dev/null 2>&1; then
+        echo "* Trento is already installed. Updating trento"
+        zypper up -y trento > /dev/null
+    else
+        echo "* Installing trento"
+        zypper in -y trento > /dev/null
+    fi
+}
+
+install_consul
+setup_consul
+install_trento
+
+echo -e "\e[92mDone.\e[97m"
+echo -e "You can now start trento-agent with: \033[1msystemctl start trento-agent\033[0m"
```

Comment:
I would say, "Now you can start the trento-agent with", what @lee-martin would say?

Reply:
yes it sounds better

Reply:
agree on "Now you can start the trento-agent with". thanks.